dhall-nix/exec/Main.hs

```diff
@@ -48,14 +48,20 @@ main = handle (Dhall.detailed (do
         Left  err  -> Control.Exception.throwIO err
         Right expr -> return expr
 
-    expr' <- Dhall.Import.load expr
-    case Dhall.TypeCheck.typeOf expr' of
-        Left  err -> Control.Exception.throwIO err
-        Right _   -> return ()
+    -- expr' <- Dhall.Import.load expr
+    -- case Dhall.TypeCheck.typeOf expr' of
+    --     Left  err -> Control.Exception.throwIO err
+    --     Right _   -> return ()
```

Comment:
It is unfortunate that I had to comment out the type-checking here.  I couldn't find any functions in `Dhall.TypeCheck` that would allow type-checking on Dhall expressions that still have imports (which makes sense).

Maybe it would make sense to change the implementation in this PR to instead be more like `dhall-nixpkgs`?  I could use the similar logic to `dhall-nixpkgs` to search through the Dhall expression tree for remote imports, and use FODs to load them into a cache.  I could then make use of this cache when evaluating my main Dhall file, similar to how `buildDhallDirectoryPackage` works.

I'm guessing this would also let me type-check the input Dhall file, since all of its remote url imports will be in the cache.